dkpro-core-io-brat-asl/src/test/java/org/dkpro/core/io/brat/BratReaderWriterTest.java

```diff
@@ -300,4 +629,118 @@ public void testBratWithDiscontinuousFragmentFar()
 
     @Rule
     public DkproTestContext testContext = new DkproTestContext();
+    
+    ////////////////////////////////////////////////////////////
+    // Alain Desilets
+    //
+    //  To test my improvements, I could not use testOneWay
+    //  and testRoundtrip.
+    //
+    //  The reason is that these methods do WAY too much and in
+    //  particular, they "patch" the reader description to avoid
+    //  problems caused by the very situations I am trying to 
+    //  deal with. In other words, if I use testOneWay to test 
+    //  a situation like "forgetting to add *.ann at the end of
+    //  a directory", the test will succeed because testOneWay() 
+    //  adds *.ann if not present.
+    //
+    //  So I created a simpler runner for tests called 
+    //  testReadWrite().
+    ////////////////////////////////////////////////////////////
+    
+    public static class JCasCollector extends JCasAnnotator_ImplBase {
```

Comment:
A JCasCollector should normally not be needed. Mind that the CAS object is normally re-used and adding it to a list would only repeatedly add the same object over and over. If it is necessary to collect multiple CASes, then code such as this needs to be used:

```
List<JCas> cases = new ArrayList
CollectionReader reader = createReader(...)
while (reader.hasNext()) {
  JCas jcas = JCasFactory.createJCas();
  reader.getNext(jcas);  
  cases.add(jcas);
}
```